frontend/src/lib/fabricCanvasUtils.ts

```diff
@@ -0,0 +1,364 @@
+};
+
+
+
+export const handleRemoveSelected = (canvas: fabric.Canvas | null): void => {
+  if (canvas) {
+    const activeObjects = canvas.getActiveObjects();
+    activeObjects.forEach((obj) => canvas.remove(obj));
+    canvas.discardActiveObject().requestRenderAll();
+  }
+};
+
+export const handleGroupSelected = (canvas: fabric.Canvas | null): void => {
+  if (canvas) {
+    const activeObjects = canvas.getActiveObjects();
+    if (activeObjects.length) {
+      const clonedObjects = activeObjects.map((obj) =>
+        fabric.util.object.clone(obj)
+      );
+
+      const group = new fabric.Group(clonedObjects, {
+        originX: "center",
+        originY: "center",
+      });
+
+      activeObjects.forEach((obj) => canvas.remove(obj));
+      canvas.add(group);
+      canvas.setActiveObject(group);
+      canvas.requestRenderAll();
+    }
+  }
+};
+
+export const addLine = (canvas: fabric.Canvas | null, options?: { color?: string; strokeWidth?: number }): boolean => {
+  const { color = "black", strokeWidth = 5 } = options || {};
+  const line = new fabric.Line([50, 100, 200, 200], {
+    stroke: color,
+    strokeWidth: strokeWidth,
+    selectable: true,
+  });
+  canvas?.add(line);
+  canvas?.setActiveObject(line);
+  return false;
+};
+
+export const addRectangle = (canvas: fabric.Canvas | null, options?: { fillColor?: string; width?: number; height?: number }): void => {
+  const { fillColor = "red", width = 200, height = 100 } = options || {};
+  const rect = new fabric.Rect({
+    left: 100,
+    top: 100,
+    fill: fillColor,
+    width: width,
+    height: height,
+    angle: 0,
+    transparentCorners: false,
+  });
+  canvas?.add(rect);
+  canvas?.setActiveObject(rect);
+};
+
+export const addCircle = (canvas: fabric.Canvas | null, options?: { fillColor?: string; radius?: number }): void => {
+  const { fillColor = "green", radius = 50 } = options || {};
+  const circle = new fabric.Circle({
+    radius: radius,
+    fill: fillColor,
+    left: 200,
+    top: 200,
+  });
+  canvas?.add(circle);
+  canvas?.setActiveObject(circle);
+};
+
+const removeImagesFromCanvas = (canvas:fabric.Canvas | null) => {
+  const imageObjects = canvas?.getObjects('image') as fabric.Image[];
+  const imagesData = imageObjects.map((imageObj) => ({
+    src: imageObj.getSrc(),
+    left: imageObj.left,
+    top: imageObj.top,
+    scaleX: imageObj.scaleX,
+    scaleY: imageObj.scaleY,
+    angle: imageObj.angle,
+  }));
+  imageObjects.forEach((imageObj) => canvas?.remove(imageObj));
+  canvas?.renderAll();
+  return imagesData;
+};
+
+const restoreImagesToCanvas = (canvas:fabric.Canvas | null, imagesData:any) => {
+  imagesData.forEach((imgData:any) => {
+    fabric.Image.fromURL(imgData.src, (img) => {
+      img.set({
+        left: imgData.left,
+        top: imgData.top,
+        scaleX: imgData.scaleX,
+        scaleY: imgData.scaleY,
+        angle: imgData.angle,
+      });
+      canvas?.add(img);
+      canvas?.renderAll();
```

Comment:
Do we have to rerender for every image? Is rerendering after everything is done a viable option?